### PR TITLE
pkg/build/client: remove this cruft

### PR DESCRIPTION
**Pull request by mfojtik**
Use the real client instead. This is removing and replacing obvious cruft. I will work on listers in the next episode.

/cc @deads2k

### CODE REVIEW

**Comment by deads2k** [comment]
if it passes

/lgtm

File: pkg/build/controller/build/build_controller.go [hunk 1]
```diff
@@ -36,20 +34,23 @@ import (
 	configv1 "github.com/openshift/api/config/v1"
 	imagev1 "github.com/openshift/api/image/v1"
 	buildv1client "github.com/openshift/client-go/build/clientset/versioned"
+	buildclientv1 "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
```
Comment (soltysh) [inline pkg/build/controller/build/build_controller.go]: This is bad, we're importing 2 similar stuff, I'd prefer we merge usage of the two.

File: pkg/build/controller/buildconfig/buildconfig_controller.go [hunk 2]
```diff
@@ -21,11 +21,12 @@ import (
 
 	buildv1 "github.com/openshift/api/build/v1"
 	buildclient "github.com/openshift/client-go/build/clientset/versioned"
+	buildclientv1 "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
```
Comment (soltysh) [inline pkg/build/controller/buildconfig/buildconfig_controller.go]: Ditto.

File: pkg/build/controller/build/build_controller.go [hunk 3]
```diff
@@ -2171,7 +2172,7 @@ func getBuildName(pod metav1.Object) string {
 
 func makeBuildPodOwnerRef(buildPod *corev1.Pod) metav1.OwnerReference {
 	return metav1.OwnerReference{
-		APIVersion: "v1",
+		APIVersion: "buildclientv1",
```
[the remark this answers is not in the document]
Comment (mfojtik) [reply pkg/build/controller/build/build_controller.go]: uups, good catch... goland seems to be smarter than it should  be

[the remark this answers is not in the document]
Comment (mfojtik) [reply pkg/build/controller/build/build_controller.go]: fixed.

**Pull request merged**